File: jobs/2026-04-23__10-52-17/b2b_org_settings__B9DwCSg/artifacts/update_org.js
const stytch = require('stytch');

const organizationId = process.argv[2];

if (!organizationId) {
  console.error('Usage: node update_org.js <organization_id>');
  process.exit(1);
}

const projectId = process.env.STYTCH_PROJECT_ID;
const secret = process.env.STYTCH_SECRET;

if (!projectId || !secret) {
  console.error('Missing STYTCH_PROJECT_ID or STYTCH_SECRET environment variable');
  process.exit(1);
}

const client = new stytch.B2BClient({
  project_id: projectId,
  secret,
});

async function updateOrganization() {
  await client.organizations.update({
    organization_id: organizationId,
    auth_methods: 'RESTRICTED',
    allowed_auth_methods: ['sso', 'magic_link'],
    email_jit_provisioning: 'RESTRICTED',
  });

  console.log('Organization updated successfully');
}

updateOrganization().catch((error) => {
  console.error('Failed to update organization:', error);
  process.exit(1);
});

File: jobs/2026-04-26__00-27-20/b2b_update_organization__V2fUFaB/artifacts/update_org.py
import os
import stytch

def main():
    project_id = os.environ.get("STYTCH_B2B_PROJECT_ID")
    secret = os.environ.get("STYTCH_B2B_SECRET")
    
    if not project_id or not secret:
        print("Missing STYTCH_B2B_PROJECT_ID or STYTCH_B2B_SECRET")
        return

    client = stytch.B2BClient(
        project_id=project_id,
        secret=secret,
    )

    with open("/logs/artifacts/trial_id", "r") as f:
        trial_id = f.read().strip()

    org_name = f"test-org-{trial_id}"
    org_slug = f"test-org-{trial_id}"

    # Create organization
    create_resp = client.organizations.create(
        organization_name=org_name,
        organization_slug=org_slug,
    )
    
    org_id = create_resp.organization.organization_id
    
    # Update organization
    update_resp = client.organizations.update(
        organization_id=org_id,
        auth_methods="RESTRICTED",
        allowed_auth_methods=["sso", "magic_link"]
    )
    
    print(f"Successfully created and updated organization: {org_id}")

if __name__ == "__main__":
    main()
